### PR TITLE
[SRVCOM-1770][RELEASE-1.3] Bump prom client to 1.11.1 for CVE-2022-21698

### PR DESCRIPTION
**Pull request by skonto**
This is needed in 1.3 (1.24) only, it is fixed on branch release-1.4: https://github.com/openshift-knative/net-kourier/blob/release-1.4/go.sum#L507. We might still have to add the replacement directive on 1.4+ though since there is a question whether the downstream security tool is smart enough to pick the [latest version in go.sum file](https://stackoverflow.com/questions/69825533/why-does-go-sum-include-so-many-older-packages).

### CODE REVIEW

**Comment by skonto** [comment]
@rhuss @nak3 need a lgtm or close this. Others repos merged this, if we close please not that will need to clarify in release notes about false positives.

File: go.mod [hunk 1]
```diff
@@ -23,3 +23,6 @@ require (
 	knative.dev/networking v0.0.0-20220302134042-e8b2eb995165
 	knative.dev/pkg v0.0.0-20220301181942-2fdd5f232e77
 )
+
+// Enforce a version for CVE-2022-21698
```
Comment (rhuss) [inline go.mod:27]: Thanks for adding the comment. Maybe add also that this line can be removed in the future and what the condition needs to be fulfilled ? I.e. which direct dependency needs to be monitored, that eventually attracts the prometheus dep.

**Comment by rhuss** [comment]
+1 for merging this to keep prodsec happy (although I know that it does not really affect us, but as long there is no possibility to communicate this back to the sec scan tooling via some annotation, let's express our knowledge here).

But maybe extend the comment a bit more like proposed above.

**Comment by rhuss** [comment]
One check is failing, too.

File: go.mod (hunk 1 above)
Comment (rhuss) [reply go.mod:27]: Please also add, that we do not actively use this dependency but this has been added to keep the sec scanning tools happy.

**Comment by skonto** [comment]
Codegen fails on net-kourier, net-istio repos because we run the upstream github actions without modifications (we dont apply the patches again after codegen in that action, the action knows nothing about patches).

**Pull request merged**